### ISSUE TITLE
Failure to propagate `HOST` environment variable from compiletest_rs's build script to crate build on x86_64

### ISSUE THREAD

**Issue by thomcc** [mozilla/sccache#1766]
Hi. We're getting the following error while building `compiletest_rs` under sccache on x86_64:

```
error: environment variable `HOST` not defined
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/compiletest_rs-0.9.0/src/common.rs:397:19
    |
397 |             host: env!("HOST").to_string(),
    |                   ^^^^^^^^^^^^
    |
    = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `compiletest_rs` due to previous error
```

This should be set by `compiletest_rs`'s build script: https://github.com/Manishearth/compiletest-rs/blob/master/build.rs#L6, but doesn't seem to be when building for x86_64 (perplexingly, it works for aarch64).

*Potentially irrelevant background: In https://github.com/tcdi/plrust we're trying to introduce sccache into our build system (CC @BradyBonnette who is actually doing most of the work here). As part of that project, we have a custom rustc_driver-user that uses uses [the `compiletest_rs` crate](https://github.com/Manishearth/compiletest-rs) crate to test itself (usage is in <https://github.com/tcdi/plrust/blob/main/plrustc/plrustc/tests/uitests.rs>, in case that helps you find the actual code and such).*

**Comment by thomcc** [comment]
Hmm, that's unfortunate. So compiletest-rs can't be used with sccache then, unless it's fixed.

**Comment by drahnr** [comment]
I think removing host _might_ have been an overshoot, but I am not sure in which contexts it's used. I definitely could see potential for abuse though, especially in the C/CXX domain.

**Comment by drahnr** [comment]
Note that I have to double check, if i.e. `cargo:rerun-if-env-changed=HOST` would re-include it from a `build.rs` and if that'd be applicable in your case.

**Comment by thomcc** [comment]
`HOST` is one of the vars cargo sets for build scripts: https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts, so generally rerun-if-env-changed shouldn't be used for it (but perhaps sccache changes this?)

**Comment by drahnr** [comment]
I think the patch originates from an environment where`HOST` is used to denote the _name_ of the host where a rpm package is built (I don't quite remember the context) and hence was a source of randomness. Now `build.rs` is a different story. It'd be good to be able to discriminate between build script and regular binary, but I did not investigate that.

**Comment by thomcc** [comment]
Yeah, so cargo sets HOST when running build.rs's. This build.rs then propagates it to the actual build of the crate, but it's not picked up.

**Comment by drahnr** [comment]
I'd be very happy to review a PR in that direction, it's non trivial though and definitely requires some digging in how cargo handles the different targets. I won't have time to investigate anytime soon.

**Comment by thomcc** [comment]
It's probably easier to fix compiletest-rs in this case: https://github.com/Manishearth/compiletest-rs/pull/269

That said I would say that stripping HOST in this manner is almost certainly overeager, given that much more rust code will care about it in this case.

**Comment by thomcc** [comment]
I fixed this in compiletest-rs, so I renamed the issue.
